src/backend/distributed/connection/connection_management.c

```diff
@@ -995,6 +995,13 @@ StartConnectionEstablishment(ConnectionHashKey *key)
 	ConnParamsHashEntry *entry = hash_search(ConnParamsHash, key, HASH_ENTER, &found);
 	if (!found || !entry->isValid)
 	{
+		if (!found)
```

Comment:
If `MemoryContextAllocZero` throws an error we don't want the next lookup to call `FreeConnParamsHashEntryFields` with potentially random pointers

Reply:
Can you put that into a comment?